drivers/gles3/storage/texture_storage.cpp

```diff
@@ -1509,8 +1509,19 @@ void TextureStorage::_texture_set_data(RID p_texture, const Ref<Image> &p_image,
 	GLenum internal_format;
 	bool compressed = false;
 
+	bool needs_decompress = texture->resize_to_po2;
+
+#ifdef WEB_ENABLED
```

Comment:
After looking into it, this seems to be true for all GLES3 contexts since they rely on [`EXT_texture_compression_rgtc `](https://registry.khronos.org/OpenGL/extensions/EXT/EXT_texture_compression_rgtc.txt) which explicitly does not support 3D textures. 

Therefore, I think this needs to use `is_gles_over_gl()` instead to capture any time we use a GLES3 context (i.e. web, mobile, and ANGLE on desktop). 

I am not 100% sure about ANGLE on desktop though, it might work anyway despite not being guaranteed by the spec